cmd/caa-log-checker/main.go

```diff
@@ -79,19 +80,37 @@ func checkIssuances(scanner *bufio.Scanner, checkedMap map[string][]time.Time, s
 		}
 
 		var badNames []string
+		var timeErrors []float64
+
 		for _, name := range ie.Names {
 			nameOk := false
+
+			var minTimeError float64 = math.Inf(+1)
+
 			for _, t := range checkedMap[name] {
-				if t.Before(ie.issuanceTime) && t.After(ie.issuanceTime.Add(-8*time.Hour)) {
+				validStart := ie.issuanceTime.Add(-8 * time.Hour)
+				validEnd := ie.issuanceTime
+				if t.After(validStart) && t.Before(validEnd.Add(timeTolerance)) {
 					nameOk = true
+				} else {
+					// If the check didn't pass, calculate how much tolerance we'd need for it to
+					// pass, to make it easier to diagnose log timestamp desync.
+					if !t.Before(validEnd) {
```

Comment:
Can you collapse this into an `else if`?

Also the condition here is not quite exactly the same as the `t.Before(validEnd.Add(timeTolerance))` condition above. That makes this a little confusing to reason about.

Here's an idea to clarify:

```
if t.After(validStart) && t.Before(validEnd.Add(timeTolerance)) {
} else if t.After(validStart) {
```

This solves a couple things: There's a clear parallel with the first clause of the first if, so the reader can conclude this gets executed only if the second clause failed. And also we remove the double-negative-ish `!t.Before` (which would be simpler to express as `t.After` except that slightly fudges the case where things are equal).